ebpf_ffi/ebpf.cc

```diff
@@ -83,25 +96,15 @@ struct bpf_result ffi_load_ebpf_program(void *serialized_proto, size_t size,
                                         int coverage_enabled,
                                         uint64_t coverage_size) {
   std::string verifier_log, error_message;
-  std::string serialized_proto_string(
-      reinterpret_cast<const char *>(serialized_proto), size);
-  EncodedProgram program;
-  if (!program.ParseFromString(serialized_proto_string)) {
-    error_message = "Could not parse EncodedProgram proto";
-  }
 
   struct coverage_data cover;
   memset(&cover, 0, sizeof(struct coverage_data));
   cover.fd = -1;
   cover.coverage_size = coverage_size;
   if (coverage_enabled) enable_coverage(&cover);
-  uint8_t *prog_buff = (uint8_t *)(program.program().c_str());
-  uint8_t *btf_buff = (uint8_t *)(program.btf().c_str());
-  uint8_t *func_buff = (uint8_t *)(program.function().c_str());
 
-  int program_fd = load_ebpf_program(
-      prog_buff, program.program().length(), btf_buff, program.btf().length(),
-      func_buff, program.function().length(), &verifier_log, &error_message);
+  int program_fd =
+      load_ebpf_program(serialized_proto, size, &verifier_log, error_message);
```

Comment:
should we de serialize the proto in this function so the load_ebpf_program doesn't have to do it?

Reply:
Done